tests/controllers/villains.test.js

```diff
@@ -0,0 +1,141 @@
+const chai = require('chai')
+const sinon = require('sinon')
+const sinonChai = require('sinon-chai')
+const models = require('../../models')
+const {
+  after, afterEach, before, beforeEach, describe, it
+} = require('mocha')
+const { villainsList, singleVillain } = require('../mocks/villains')
+const { getAllVillains, getVillainBySlug, saveNewVillain } = require('../../controllers/villains.js')
+
+chai.use(sinonChai)
+const { expect } = chai
+
+describe('Controllers - villains', () =>
+{
+  let sandbox
+  let stubbedFindAll
+  let stubbedFindOne
+  let stubbedCreate
+  let stubbedSend
+  let response
+  let stubbedSendStatus
+  let stubbedStatusDotSend
+  let stubbedStatus
+
+  before(() =>
+  {
+    sandbox = sinon.createSandbox()
+    stubbedFindAll = sandbox.stub(models.villains, 'findAll')
+    stubbedFindOne = sandbox.stub(models.villains, 'findOne')
+    stubbedCreate = sandbox.stub(models.villains, 'create')
+    stubbedSend = sandbox.stub()
+    stubbedSendStatus = sandbox.stub()
+    stubbedStatusDotSend = sandbox.stub()
+    stubbedStatus = sandbox.stub()
+    response = {
+      send: stubbedSend,
+      sendStatus: stubbedSendStatus,
+      status: stubbedStatus,
+    }
+  })
+  beforeEach(() =>
+  {
+    stubbedStatus.returns({ send: stubbedStatusDotSend })
+  })
+  afterEach(() =>
+  {
+    sandbox.reset()
+  })
+  after(() =>
+  {
+    sandbox.restore()
+  })
+  describe('getAllVillains', () =>
+  {
+    it('Retrieves a list of villains and calls response.send() with the list.', async () =>
+    {
+      stubbedFindAll.returns(villainsList)
+      await getAllVillains({}, response)
+      expect(stubbedFindAll).to.have.callCount(1)
+      expect(stubbedFindAll).to.have.been.calledWith({ attributes: ['name', 'movie', 'slug'] })
+      expect(stubbedSend).to.have.been.calledWith(villainsList)
+    })
+    it('Returns a 500 error with an error message when the database call throws an error.', async () =>
+    {
+      stubbedFindAll.throws('ERROR!')
+      await getAllVillains({}, response)
+      expect(stubbedFindAll).to.have.callCount(1)
+      expect(stubbedStatus).to.have.been.calledWith(500)
+      expect(stubbedStatusDotSend).to.have.been.calledWith('Unable to retrieve all villains try again.')
```

Comment:
This is a silly failure. The message that you use on line 10 of the controller is not the same as here. This expect and the one above it are asserting that the code properly calls the catch code when an error is thrown, that means calling line 10 from your controller file.

```JavaScript
return response.status(500).send('Cannot retrieve villains please try again')
```

This means that you need to expect that status was called with 500 because `response.status(500)` and that the send is called with `'Cannot retrieve villains please try again'` because that is what is being passed in